<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
    "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html  xmlns="http://www.w3.org/1999/xhtml" xml:lang="ja" lang="ja">
<!-- generated by to_html.pl from f_System.xml -->
<head>
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
	<title>exceptionHandler - Handler function for uncaught exception</title>
	<meta name="author" content="W.Dee" />
	<meta http-equiv="Content-Style-Type" content="text/css" />
	<meta http-equiv="Content-Script-Type" content="text/javascript" />
	<link href="funcref.css" type="text/css" rel="stylesheet" title="クラスリファレンス用標準スタイル" />
	<link href="mailto:dee@kikyou.info" rev="Made" />
	<link href="index.html" target="_top" rel="Start" title="トップページ" />
	<link href="f_System.html" title="Systemクラス" rel="Parent" />
</head>
<body>
<h1><span class="fheader"><a name="top" id="top">System.exceptionHandler</a></span></h1><div class="para"><div>
<dl>
<dt>機能/意味</dt>
<dd>
Handler function for uncaught exception</dd>
<dt>タイプ</dt>
<dd>
<a class="jump" href="f_System.html">Systemクラス</a>のプロパティ (読み書き可能)<br />
</dd>
<dt>説明</dt>
<dd>
		　Represents a function that handles uncaught exceptions (exceptions passed to Kirikiri core without being caught anywhere).<br />
		　Specifying null results in the default behavior.<br />
		　The default behavior is the following:
		<ol>			<li>Stop delivery of asynchronous events (set <a class="jump" href="f_System_eventDisabled.html">System.eventDisabled</a> to true)</li>			<li>Start outputting log to file (call <a class="jump" href="f_Debug_logAsError.html">Debug.logAsError</a>)</li>			<li>Displays a dialog box that informs you of the error, and indicates the location in the script editor</li>		</ol>		　<br />
		　The handler function takes one argument, which is the exception object.<br />
		　If no handler function is specified, if the handler function is null, or if the handler function returns false, the default action will be taken.<br />
		　If the handler function returns true, the above default action will not be taken.<br />
		　Consider the possibility of asynchronous events occurring while executing the handler function. If the Kirikiri core can handle the asynchronous event, an unexpected exception may occur again while executing the exception handler. In order to avoid this, usually, when performing processing that waits for something in the handler function (when Kirikiri has a chance to process the asynchronous event), stop the occurrence of the asynchronous event.<br />
		<code class="bq"><span class="weak">例:</span><br />System.exceptionHandler&nbsp;=&nbsp;function&nbsp;(e)<br />
{<br />
&nbsp;&nbsp;&nbsp;&nbsp;//&nbsp;This&nbsp;function&nbsp;is&nbsp;called&nbsp;when&nbsp;an&nbsp;uncaught&nbsp;exception&nbsp;is&nbsp;caught&nbsp;by&nbsp;the&nbsp;system.<br />
&nbsp;&nbsp;&nbsp;&nbsp;//&nbsp;e&nbsp;is&nbsp;the&nbsp;exception&nbsp;object.<br />
&nbsp;&nbsp;&nbsp;&nbsp;if(e&nbsp;instanceof&nbsp;&quot;ConductorException&quot;)<br />
&nbsp;&nbsp;&nbsp;&nbsp;{<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;//&nbsp;For&nbsp;exceptions&nbsp;thrown&nbsp;by&nbsp;conductors<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Debug.logAsError();&nbsp;//&nbsp;Start&nbsp;writing&nbsp;log&nbsp;to&nbsp;file,&nbsp;etc.<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;var&nbsp;event_disabled&nbsp;=&nbsp;System.eventDisabled;<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;System.eventDisabled&nbsp;=&nbsp;true;<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;//&nbsp;Stop&nbsp;the&nbsp;event&nbsp;once&nbsp;because&nbsp;it&nbsp;is&nbsp;troublesome&nbsp;if&nbsp;an&nbsp;event&nbsp;occurs&nbsp;while&nbsp;displaying&nbsp;the&nbsp;reason&nbsp;of&nbsp;the&nbsp;error<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;System.inform(e.message);<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;System.eventDisabled&nbsp;=&nbsp;event_disabled;<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;//&nbsp;Whether&nbsp;or&nbsp;not&nbsp;to&nbsp;fire&nbsp;the&nbsp;event<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;true;&nbsp;//&nbsp;If&nbsp;true&nbsp;is&nbsp;returned,&nbsp;exception&nbsp;processing&nbsp;will&nbsp;not&nbsp;be&nbsp;performed&nbsp;on&nbsp;the&nbsp;core&nbsp;side<br />
&nbsp;&nbsp;&nbsp;&nbsp;}<br />
&nbsp;&nbsp;&nbsp;&nbsp;else<br />
&nbsp;&nbsp;&nbsp;&nbsp;{<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;return&nbsp;false;&nbsp;//&nbsp;Normal&nbsp;exception&nbsp;handling&nbsp;when&nbsp;returning&nbsp;false<br />
&nbsp;&nbsp;&nbsp;&nbsp;}<br />
};<br />
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</code>
	</dd>
<dt>参照</dt>
<dd>
<a class="jump" href="f_System_eventDisabled.html">System.eventDisabled</a><br />
<a class="jump" href="f_Debug_logAsError.html">Debug.logAsError</a><br />
</dd>
</dl>
</div></div>
	<script type="text/javascript" charset="UTF-8" src="documentid.js" ></script>
	<script type="text/javascript" charset="UTF-8" src="postcontent.js" ></script>
</body>
</html>
